libs/remixd/src/bin/remixd.ts

```diff
@@ -38,15 +38,24 @@ import * as program from 'commander'
     console.log('\x1b[33m%s\x1b[0m', '[WARN] Any application that runs on your computer can potentially read from and write to all files in the directory.')
     console.log('\x1b[33m%s\x1b[0m', '[WARN] Symbolic links are not forwarded to Remix IDE\n')
     try {
-      const sharedFolderClient = new servicesList.Sharedfolder()
-      const websocketHandler = new WebSocket(65520, { remixIdeUrl: program.remixIde }, sharedFolderClient)
+      // shared folder
+      const websocketHandler = new WebSocket(65520, { remixIdeUrl: program.remixIde }, () => new servicesList.Sharedfolder())
 
-      websocketHandler.start((ws: WS) => {
+      websocketHandler.start((ws: WS, sharedFolderClient: servicesList.Sharedfolder) => {
         sharedFolderClient.setWebSocket(ws)
         sharedFolderClient.setupNotifications(program.sharedFolder)
         sharedFolderClient.sharedFolder(program.sharedFolder, program.readOnly || false)
       })
       killCallBack.push(websocketHandler.close.bind(websocketHandler))
+
+      // git
+      const websocketHandlerForGit = new WebSocket(65521, { remixIdeUrl: program.remixIde }, () => new servicesList.GitClient())
```

Comment:
Look like code duplication. If remixd is supposed to manage multiple port build it up for that : 
```javascript
const servicesList = {
  git: GitClient,
  folder: Sharedfolder
}
...
const ports = {
  git: 65521,
  folder: 65520
}
function startService<S extends 'git' | 'folder'>(service: S, client: servicesList[T]) {
  const socket = new Websocket(port[service], new servicesList[service]());
  socket.start((...) => ...)
}
```